### PR TITLE
Add hardware requirements to tutorial

### PR DESCRIPTION
**Pull request by ghislainbourgeois**
# Description

Add hardware requirements to tutorial. The requirements were estimated from running the tutorial locally. The CPU requirements are for the `UPF` container built for recent CPUs.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [ ] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

**Review by saltiyazan — APPROVED** [review]

File: docs/tutorial/index.md [hunk 1]
```diff
@@ -16,6 +16,13 @@ Charmed 5G - it will save you many more hours later on. Follow the
 tutorial steps in sequence; they take you on a learning journey through the
 product.
 
+To complete this tutorial, you will need a machine with the following
+requirements:
+
+- A recent `x86_64` CPU with the `AVX-512` feature.
```
Comment (gruyaume) [inline docs/tutorial/index.md:22]:
```suggestion
- A `x86_64` CPU with the `AVX-512` feature (Intel 8th gen or more recent)
```

File: docs/tutorial/index.md [hunk 2]
```diff
@@ -16,6 +16,13 @@ Charmed 5G - it will save you many more hours later on. Follow the
...
+- 8Gb of RAM
```
Comment (gruyaume) [inline docs/tutorial/index.md:23]:
```suggestion
- 8GB of RAM
```

File: docs/tutorial/index.md [hunk 3]
```diff
@@ -16,6 +16,13 @@ Charmed 5G - it will save you many more hours later on. Follow the
...
+- 50Gb of free disk space
```
Comment (gruyaume) [inline docs/tutorial/index.md:24]:
```suggestion
- 50GB of free disk space
```

File: docs/tutorial/index.md (hunk 1 above)
Comment (ghislainbourgeois) [reply docs/tutorial/index.md:22]: This is now a bit misleading unfortunately. On the consumer market, most 12th gen will have the feature, but not the later production ones. 13th gen does not have it. On AMD's side, the Zen4 architecture seems to support it.

On the server market, it is a lot more true, but I think a lot of people will run this tutorial on a laptop or desktop, probably in a VM.

I just tried to validate on my laptop, and it seems I do not have the feature, on a 12th gen. I might be wrong with the feature set required, I will try to narrow it a bit further.

Thanks for the comment.

Comment (ghislainbourgeois) [reply docs/tutorial/index.md]: I clarified the requirements for both Intel and AMD.

**Pull request merged**